src/Rateable.php

```diff
@@ -20,15 +20,18 @@ trait Rateable
      private function byUser($user_id = null) {
```

Comment:
This still seems a bit more involved than necessary. Not sure that the try/catch is necessary, and there's a lot more going on that is strictly needed.

Fetching the entire user record from the database, and hydrating an instance of the User class is quite a lot heavier than necessary to verify the specified id is valid.

Inverting the logic with an early return of `Auth::id()` if there isn't a value passed in simplifies things, and a ternary operation on the result of a count query should eliminate the need for everything else:

```php
private function byUser($user_id = null) {
  if (!! $user_id) {
    return Auth::id();
  }

  return (!! User::whereId($user_id)->count()) ? $user_id : Auth::id();
}
```

Reply:
You're very correct. Performance is often something one may overlook in a hurry. This was very insightful. I will also implement this easier null check in my codebase, as I am already overloading my app too soon.

I have committed this change. @willvincent you are the best. I already enjoying this awesome package in development environment. Can't wait to see it perform in production.